app/services/create_newsletter_signup.rb

```diff
@@ -0,0 +1,45 @@
+require 'sendgrid-ruby'
+include SendGrid
+
+class CreateNewsletterSignup
+  attr_accessor :newsletter_signup, :first_name, :last_name, :email
+
+  def initialize(first_name:, last_name:, email:)
+    @newsletter_signup = {
+      first_name: first_name,
+      last_name: last_name,
+      email: email
+    }
+  end
+
+  def perform
+    if @newsletter_signup[:first_name].blank? || @newsletter_signup[:last_name].blank? || @newsletter_signup[:email].blank?
```

Comment:
No need to validate here. Ruby does that for you! If any of those KW args are not passed, you should see an argument error. You can test this too!

Reply:
@ColinW520 That's what I was going for initially but I was unable to get it to throw an argument error when an empty string is passed from the form. I'm searching for something I can add so it will not accept an empty string but not finding much, it only throws the error when a nil argument is passed.